gpflow/base.py

```diff
@@ -307,6 +304,16 @@ def _run_op(a, *args, **kwargs):
 Parameter._OverloadAllOperators()
 tf.register_tensor_conversion_function(Parameter, lambda x, *args, **kwds: x.read_value())
 
+TensorLike = Union[np.ndarray, tf.Tensor, tf.Variable, Parameter]
+"""
+Type alias for tensor-like types that are supported by most TensorFlow, NumPy and GPflow operations.
+
+NOTE: Does not work with multipledispatch dispatchers. See `TensorLikeTypes`.
+"""
```

Comment:
```suggestion
# Type alias for tensor-like types that are supported by most TensorFlow, NumPy and GPflow operations.
#
# NOTE: The multipledispatch package uses tuples, not typing.Union, for dispatching on multiple types.
# For calling a Dispatcher's register() method (e.g. the `@conditional.register` decorator), use `TensorLikeTypes` instead
```

Reply:
i chose docstrings so that it would appear in the docs. users may want to use these types in their own code

Reply:
Will we have to add `autodata` or some such to the API documentation for this to actually get pulled in?
I just looked it up as I didn't know it handled module-constant docstrings like this; Sphinx also supports `#:` instead which I think might be more readable when just reading the source code?

Reply:
meh - is that needed here?